contrib/mpi-proxy-split/split_process.cpp

```diff
@@ -247,12 +247,10 @@ read_lh_proxy_bits(pid_t childpid)
   // NOTE:  This requires same privilege as ptrace_attach (valid for child).
   //        Anecdotally, in containers, we've seen a case where this errors out
   //        with ESRCH (no such proc.); it may need CAP_SYS_PTRACE privilege??
-  for (int i = 0; i < IOV_SZ; i++) {
-    JTRACE("Reading segment from lh_proxy")
-          (remote_iov[i].iov_base)(remote_iov[i].iov_len);
-    ret = process_vm_readv(childpid, remote_iov + i, 1, remote_iov + i, 1, 0);
-    JASSERT(ret != -1)(JASSERT_ERRNO).Text("Error reading data from lh_proxy");
-  }
+  JTRACE("Reading segment from lh_proxy")
+         (remote_iov[0].iov_base)(remote_iov[0].iov_len);
+  ret = process_vm_readv(childpid, remote_iov, IOV_SZ, remote_iov, IOV_SZ, 0);
```

Comment:
@rohgarg @gc00 @xuyao0127 Not sure if I understand it correctly, I tested it and works fine.
Please comment if there is anything I missed.

Reply:
What's the intent here?

Reply:
I think it would be better to take the advantage of `process_vm_readv`?
I know it doesn't have too much difference given `IOV_SZ` is 2 here, I think using the standard way probably performs better than we do the loop here when `IOV_SZ` is large.